### PR TITLE
Added the round( ) function to avoid error while multiplication of float numbers issue: #46362

### LINKED ISSUE #46362: BUG: The indexes of `DataFrame.describe(percentiles=[0.29, 0.57, 0.58])` are incorrect
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])

 	value
count 	3.00
mean 	2.00
std 	1.00
min 	1.00
29.0% 	1.58
50% 	2.00
57.0% 	2.14
58.0% 	2.16
max 	3.00
```

### Issue Description

The indexes ['29.0%', '57.0%', '58.0%'] should be named as ['29%', '57%', '58%'].

### Expected Behavior

```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58]).index

# Output:
Index(['count', 'mean', 'std', 'min', '29.0%', '50%', '57.0%', '58.0%', 'max'], dtype='object')

# Expected:
Index(['count', 'mean', 'std', 'min', '29%', '50%', '57%', '58%', 'max'], dtype='object')
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.9.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.13.0-30-generic
Version          : #33-Ubuntu SMP Fri Feb 4 17:03:31 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_IL
LOCALE           : en_IL.UTF-8

pandas           : 1.4.1
numpy            : 1.22.2
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 22.0.3
setuptools       : 59.8.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 4.4.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : 2.9.3
jinja2           : 3.0.3
IPython          : 7.32.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : 2022.02.0
gcsfs            : None
matplotlib       : 3.5.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.02.0
scipy            : 1.8.0
sqlalchemy       : 1.4.31
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by ShamithaReddy**
- Added round( ) function to round the values and then convert it to int type while they are passed into the list-out, so that we can avoid conversations of int(28.999999999999996) to 28.
- if we round the value first it will be converted into 29 and then formating them to int type.
- ✅ closes #46362 
- ✅ [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- ✅ All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- ✅ Added an entry in the latest `doc/source/whatsnew/v1.4.2.rst`

### CODE REVIEW

File: doc/source/whatsnew/v1.4.2.rst [hunk 1]
```diff
@@ -30,7 +30,7 @@ Bug fixes
 ~~~~~~~~~
 - Fix some cases for subclasses that define their ``_constructor`` properties as general callables (:issue:`46018`)
 - Fixed "longtable" formatting in :meth:`.Styler.to_latex` when ``column_format`` is given in extended format (:issue:`46037`)
--
+- Fixed a case where the format of printing percentiles in not correct for the describe function in ``format.py`` (:issue:`46362`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.2.rst:33]: move to 1.5

**Review by jreback — CHANGES_REQUESTED** [review]
needs a test to replicate the original issue (the describe case)

**Comment by ShamithaReddy** [comment]
> needs a test to replicate the original issue (the describe case)
@jreback 
I did already include the original test case in https://github.com/pandas-dev/pandas/pull/46388/commits/998f9174c27d3cee9eddc90428e0e0765ee5979a

File: doc/source/whatsnew/v1.4.2.rst [hunk 2]
```diff
@@ -30,7 +30,7 @@ Bug fixes
 ~~~~~~~~~
 - Fix some cases for subclasses that define their ``_constructor`` properties as general callables (:issue:`46018`)
 - Fixed "longtable" formatting in :meth:`.Styler.to_latex` when ``column_format`` is given in extended format (:issue:`46037`)
--
+
```
Comment (datapythonista) [inline doc/source/whatsnew/v1.4.2.rst:33]: Can you revert this please?

File: pandas/tests/io/formats/test_format.py [hunk 3]
```diff
@@ -3262,6 +3262,11 @@ def test_format_percentiles():
     expected = ["0%", "50%", "2.0%", "50%", "66.67%", "99.99%"]
     assert result == expected
 
+    # Issue #46362
+    result = fmt.format_percentiles([0.281,0.57,0.58,0.29])
```
Comment (datapythonista) [inline pandas/tests/io/formats/test_format.py:3266]: You need spaces after the commas. Also, those are supposed to be percentiles, so they should be sorted.

File: doc/source/whatsnew/v1.5.0.rst [hunk 4]
```diff
@@ -362,6 +362,7 @@ Conversion
 - Bug in :meth:`Series.astype` and :meth:`DataFrame.astype` from floating dtype to unsigned integer dtype failing to raise in the presence of negative values (:issue:`45151`)
 - Bug in :func:`array` with ``FloatingDtype`` and values containing float-castable strings incorrectly raising (:issue:`45424`)
 - Bug when comparing string and datetime64ns objects causing ``OverflowError`` exception. (:issue:`45506`)
+- Bug in :meth:`format.py` the percentiles values are converted to integer dtype, which ignores the floating point error while multiplying. (:issue:`46362`)
```
Comment (datapythonista) [inline doc/source/whatsnew/v1.5.0.rst:365]: This seems a bit confusing. What about something more like "Bug when formatting percentiles, more decimals than needed shown" or something similar.

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution @ShamithaReddy. Implementation looks good. Added some comments. I think they should also leave the CI green, but would be great if you can check it yourself, and fix any other problem that make it red.

You've also got a conflict that you need to fix.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to be stale. If interested in continuing, please merge in the main branch, address the review, and we can reopen.

**Pull request closed without merging**